### DISCUSSION TITLE
Typescript plugin options example no longer accurate

### DISCUSSION THREAD

**Discussion by theoephraim** [vuejs/pinia#685] [Help and Questions] [answered]
Updating to the latest version and rereading the docs, I came across this - https://pinia.esm.dev/core-concepts/plugins.html#typing-new-creation-options

DefineStoreOptionsBase is now limited to only S and Store, which means the example, which uses the keys in the store's actions, no longer works since A is not defined.
```
declare module 'pinia' {
  export interface DefineStoreOptionsBase<S, Store> {
    debounce?: {
      // allow defining a number of ms for any of the actions
      [k in keyof A]?: number
    }
  }
}
```

Augmenting `DefineStoreOptions` still works fine. Not sure what the intention is here with having less info available for custom options?

**Comment by posva** [comment, marked as answer]
It hasn't been updated yet but you have the same information as before using some type helpers like StoreActions. It's in my todo list already 🙂

**Reply by posva** [reply]
This section has been added:

```ts
import 'pinia'

declare module 'pinia' {
  export interface DefineStoreOptionsBase<S, Store> {
    // allow defining a number of ms for any of the actions
    debounce?: Partial<Record<keyof StoreActions<Store>, number>>
  }
}
```

:::tip
There is also a `StoreGetters` type to extract the _getters_ from a Store type. You can also extend the options of _setup stores_ or _option stores_ **only** by extending the types `DefineStoreOptions` and `DefineSetupStoreOptions` respectively.
:::
